packages/lisk-p2p/src/errors.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { VError } from 'verror';
+
+export class NotEnoughPeersError extends VError {
+	private readonly code = 'NOT_ENOUGH_PEERS';
+
+	public constructor(message?: string) {
+		super(message || '');
+		this.name = 'Not Enough Peers Error';
+	}
+	public get Code(): string {
```

Comment:
Actually, I made a getter for code property because I wanted to make it read-only and private. Also, by default, this error will not have `code` property so I thought it will be a good idea to have `code`

Reply:
I think I will remove this method and make the code property public in order to avoid confusion

Reply:
Since it's an error only used in this method, I'm simply throwing `VError` with its custom name field which will act similar to what we are doing now